src/tasks/tasks.ts

```diff
@@ -75,10 +75,19 @@ function _task<P, T>(
   mockError,
   type
 ): Task<P, T> {
-  return {
+  const newTask = {
```

Comment:
any particular reason to create a temp object?

Reply:
because then I don't have to use `this` below to get the task object in the newly wrapped `TASK_RUN`.